### PR TITLE
OCPBUGS-44759: Refactor `istextorbinary` to use correct types

### PR DESCRIPTION
**Pull request by logonoff**
<h2 id="feature-fix">CONSOLE Features and Fixes:</h2>

https://issues.redhat.com/browse/OCPBUGS-44759

<h2 id="solution-description">Solution description</h2> 

Ensure buffers are used when `isBinary` is used, also prefer the named `isBinary` export over importing the entire object to allow for easier tree shaking

<h2 id="reviewers-and-assignees">Reviewers and assignees:</h2>

  Console Approver:
  /assign @TheRealJon 

<h2 id="test-cases">Test cases:</h2>

n/a

<h2 id="additional-info">Additional info:</h2>

n/a

<h2 id="screenshots">Screen shots / gifs / design review:</h2>

n/a

### CODE REVIEW

File: frontend/public/components/utils/file-input.tsx [hunk 1]
```diff
@@ -42,7 +42,7 @@ class FileInputWithTranslation extends React.Component<FileInputProps, FileInput
         ? (reader.result as string).split(',')[1]
         : (reader.result as string);
       // OnLoad, if inputFileIsBinary we have read as a binary string, skip next block
-      if (isBinary(null, input) && !fileIsBinary) {
+      if (isBinary(null, reader.result as Buffer) && !fileIsBinary) {
```
Comment (TheRealJon) [inline frontend/public/components/utils/file-input.tsx:45]: `reader.result` is not a `Buffer`. It is either a `string` or an `ArrayBuffer`, which is not the same as `Buffer`;
```suggestion
      if (isBinary(null, Buffer.from(reader.result)) && !fileIsBinary) {
```

File: frontend/public/components/utils/file-input.tsx [hunk 2]
```diff
@@ -172,7 +172,9 @@ const DroppableFileInputWithTranslation = withDragDropContext(
       this.state = {
         inputFileName: '',
         inputFileData: this.props.inputFileData || '',
-        inputFileIsBinary: this.props.inputFileIsBinary || isBinary(null, this.props.inputFileData),
+        inputFileIsBinary:
+          this.props.inputFileIsBinary ||
+          isBinary(null, Buffer.from(this.props.inputFileData, 'base64')),
```
Comment (TheRealJon) [inline frontend/public/components/utils/file-input.tsx:177]: Do we know that the input file is base64 encoded all the time?
```suggestion
          isBinary(null, Buffer.from(this.props.inputFileData)),
```

File: frontend/public/components/utils/file-input.tsx [hunk 3]
```diff
@@ -196,7 +198,7 @@ const DroppableFileInputWithTranslation = withDragDropContext(
       reader.onload = () => {
         const input = reader.result as string; // Note(Yaacov): we use reader.readAsText
         // OnLoad, if inputFileIsBinary we have read as a binary string, skip next block
-        if (isBinary(null, input) && !inputFileIsBinary) {
+        if (isBinary(null, reader.result as Buffer) && !inputFileIsBinary) {
```
Comment (TheRealJon) [inline frontend/public/components/utils/file-input.tsx:201]:
```suggestion
        if (isBinary(null, Buffer.from(reader.result)) && !inputFileIsBinary) {
```

**Review by TheRealJon — COMMENTED** [review]
I think there are a couple more places where we should be instantiating a Buffer instance rather than trying to cast. For the most part, we try to avoid type casting.

File: frontend/public/components/utils/file-input.tsx (hunk 2 above)
Comment (logonoff) [reply frontend/public/components/utils/file-input.tsx:177]: This was the change previously, however when the data is base64-encoded, `isBinary` will erroneously say that the file is text

**Comment by logonoff** [comment]
/hold

blocked due to ongoing discussion of major refactor https://issues.redhat.com/browse/CONSOLE-4088

**Comment by logonoff** [comment]
/close

PR is obsolete, will be incorporated in https://issues.redhat.com/browse/CONSOLE-4088

**Pull request closed without merging**